lib/command/src/command_factory_provider.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:paintroid/command/src/command_factory.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'command_factory_provider.g.dart';
+
+@Riverpod(keepAlive: true)
```

Comment:
Did you test this?
I'm just thinking if the providers are not disposed, but the widget that uses it is disposed, does it generate a new provider if the widget is build again or is the old one used?

Reply:
When using code generation, the providers are always [auto disposed](https://docs-v2.riverpod.dev/docs/about_code_generation) by default. Our current non-generated providers are not disposed of automatically and thus are always alive whether they have listeners attached to them or not. I marked these new providers as ```keepAlive``` because non-auto-dispose providers cannot be used with always-alive providers. In your example, the state of the provider should stay the same since it is an always-alive provider (but I have not tested it, so I am not 100% sure). In the future, when we have refactored all providers, we should reconsider which providers we should mark as ```keepAlive``` and which ones we should not.